src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/OrderPreferences/OrderPreferencesFormDataProvider.php

```diff
@@ -0,0 +1,167 @@
+ *  International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Form\Admin\Configure\ShopParameters\OrderPreferences;
+
+use PrestaShop\PrestaShop\Adapter\CMS\CMSDataProvider;
+use PrestaShop\PrestaShop\Adapter\Order\GeneralConfiguration;
+use PrestaShop\PrestaShop\Adapter\Order\GiftOptionsConfiguration;
+use PrestaShop\PrestaShop\Core\Form\FormDataProviderInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Class is responsible of managing the data manipulated using forms
+ * in "Configure > Shop Parameters > Order Settings" page.
+ */
+class OrderPreferencesFormDataProvider implements FormDataProviderInterface
+{
+    /**
+     * @var GeneralConfiguration
+     */
+    private $generalConfiguration;
+
+    /**
+     * @var GiftOptionsConfiguration
+     */
+    private $giftOptionsConfiguration;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var CMSDataProvider
+     */
+    private $cmsDataProvider;
+
+    public function __construct(
+        GeneralConfiguration $generalConfiguration,
+        GiftOptionsConfiguration $giftOptionsConfiguration,
+        TranslatorInterface $translator,
+        CMSDataProvider $cmsDataProvider
+    ) {
+        $this->generalConfiguration = $generalConfiguration;
+        $this->giftOptionsConfiguration = $giftOptionsConfiguration;
+        $this->translator = $translator;
+        $this->cmsDataProvider = $cmsDataProvider;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData()
+    {
+        return [
+            'general' => $this->generalConfiguration->getConfiguration(),
+            'gift_options' => $this->giftOptionsConfiguration->getConfiguration(),
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setData(array $data)
+    {
+        // If TOS option is disabled - reset the cms id as well
+        if (!$data['general']['enable_tos']) {
+            $data['general']['tos_cms_id'] = 0;
+        }
+
+        // If gift wrapping tax rules group was not submitted - reset it to 0
+        if (!$data['gift_options']['gift_wrapping_tax_rules_group']) {
+            $data['gift_options']['gift_wrapping_tax_rules_group'] = 0;
+        }
+
+        if ($errors = $this->validate($data)) {
+            return $errors;
+        }
+
+        return array_merge(
+            $this->generalConfiguration->updateConfiguration($data['general']),
+            $this->giftOptionsConfiguration->updateConfiguration($data['gift_options'])
+        );
+    }
+
+    /**
+     * Perform validation on form data before saving it
+     *
+     * @param array $data
+     *
+     * @return array Returns array of errors
+     */
+    protected function validate(array $data)
+    {
+        $errors = [];
+        $invalidFields = [];
+        $purchaseMinimumValue = $data['general']['purchase_minimum_value'];
```

Comment:
Maybe add some controls on `$data['general']` and `$data['gift_options']`? wdyt @mickaelandrieu ?

Reply:
yes, if these 2 keys don't exist or are null what happens?

Reply:
anyway, this can be validated by QA.